invenio_app_ils/documents/jsonresolvers/document_stock.py

```diff
@@ -32,9 +33,11 @@ def stock_resolver(document_pid):
         mediums = [
             bucket.key for bucket in search_response.aggregations.mediums.buckets
         ]
-        eitems_count = eitem_search.search_by_document_pid(document_pid).count()
-        if eitems_count > 0:
-            mediums.append("E-BOOK")
+        eitems = eitem_search.search_by_document_pid(document_pid)
+        for type in EItem.EITEM_TYPES:
+            type_count = eitems.filter("term", eitem_type=type).count()
```

Comment:
Did you check if `eitems` is immutable? When you use `.filter`, does it change the `eitems` object or simply return a new version with the filter applied?

Reply:
Yes, it returns a new version of the object with the filter applied on it, it doesn't apply it in-place